src/Symfony/Component/VarDumper/Caster/DateCaster.php

```diff
@@ -52,11 +52,14 @@ public static function castInterval(\DateInterval $interval, array $a, Stub $stu
 
     private static function formatInterval(\DateInterval $i)
     {
-        $format = '%R '
-            .($i->y ? '%yy ' : '')
-            .($i->m ? '%mm ' : '')
-            .($i->d ? '%dd ' : '')
-        ;
+        $format = '%R ';
+
+        if ($i->m >= 12 || $i->d >= 28 || $i->h >= 24 || $i->i >= 60 || $i->s >= 60) {
+            $i = date_diff($d = new \DateTime(), date_add(clone $d, $i)); // recalculate carry over points
```

Comment:
Yes, it's why the only right way to do this is to use the total number of days (``%a`` format). It's invariable even according to the current month. See the previous example https://3v4l.org/pPkNZ